src/components/LicensesFieldArray/AmendmentsFieldArray.js

```diff
@@ -35,40 +35,35 @@ class AmendmentsFieldArray extends React.Component {
     name: PropTypes.string.isRequired,
   };
 
-  warnStatusMismatch = ({ values }) => {
-    const { form, name } = this.props;
+  constructor(props) {
+    super(props);
+    this.state = { statusWarnings: [] };
 
-    get(values, name, []).forEach((field, i) => {
-      const { amendmentId, status = {} } = field;
-
-      const statusString = typeof status === 'string' ? status : status.value;
-      if (!statusString || statusString !== statuses.CURRENT) {
-        form.mutators.setFieldData(`${name}[${i}].status`, { warning: undefined });
-        return;
-      }
-
-      const amendment = get(this.props.license, 'amendments', []).find(a => a.id === amendmentId);
-
-      let warning;
-
-      // Amendment start date is in the future
-      if (new Date(amendment.startDate).getTime() > new Date().getTime()) {
-        warning = <FormattedMessage id="ui-agreements.license.warn.amendmentFuture" />;
-      }
-
-      // Amendment end date is in the past
-      if (new Date(amendment.endDate).getTime() < new Date().getTime()) {
-        warning = <FormattedMessage id="ui-agreements.license.warn.amendmentPast" />;
-      }
-
-      // Amendment has an invalid status.
-      const linkedStatus = get(amendment, 'status', {});
-      if (linkedStatus.value === statuses.EXPIRED || linkedStatus.value === statuses.REJECTED) {
-        warning = <FormattedMessage id="ui-agreements.license.warn.amendmentStatus" values={{ status: linkedStatus.label }} />;
+    if (props.items) {
+      if (props.items.length) {
+        for (let i = 0; i < props.items.length; i++) {
+          this.state.statusWarnings[i] = { warningId: '', warningValuesStatus: null };
+        }
       }
+    }
+  }
 
-      form.mutators.setFieldData(`${name}[${i}].status`, { warning });
-    });
+  RenderWarningStatus(id, valuesStatus) {
```

Comment:
Class methods should be camelCased, eg, `renderWarnings()`

Reply:
Even knowing what this PR is about, I'm not really certain what valuesStatus is going to be. After digging around I see it's the amendment record's status, which doesn't really fit this variable name. The same goes for the various other permutations of valuesStatus throughout this PR.

Reply:
Finally, I'm not really sold on the method of this function. I think the whole PR would be clearer if we cut it out.

Reply:
Yep, agreed. I wasn't super sold on this either, it seemed hacky to me, and I didn't really know what to call it. At the time it was the only way I seemed to be able to get it to work. Now I know that orders aren't changing and we can just use a simple array, this whole method has been removed.